### PR TITLE
updated tox and travis modernizer scripts to drop python 3.5 support

### PR DESCRIPTION
**Pull request by mraarif**
**Issue:** [BOM-2074](https://openedx.atlassian.net/browse/BOM-2074)

### Description

- Updated `tox_modernizer` to add `django31` in the test environments.
- Updated `travis_modernizer` to handle the following edge cases:

> 

1. Duplicated the test matrix with py38 tests (`ecommerce-worker`)
2. env error after updating the Travis file (`ecommerce-worker`)
3. Removed secure keys from env section (`credentials`)
4. Adds additional 3.8 option in python (`course-discovery`)
5. Removed env from inside include section (`course-discovery`)

### CODE REVIEW

**Review by mzulqarnain1 — APPROVED** [review]

**Comment by mraarif** [comment]
@UsamaSadiq Travis CI didn't trigger, can you look into this?

File: tests/test_travis_modernizer.py [hunk 1]
```diff
@@ -12,6 +12,8 @@ def setUp(self):
         self.test_file1 = self._setup_local_copy("test_travis.yml")
         self.test_file2 = self._setup_local_copy("test_travis_2.yml")
         self.DJANGO_PATTERN = "django[0-2][0-1]"  # creates regex to match django111, django20 && django21
+        self.allowed_versions = [3.8, '3.8']
+        self.removed_versions = [3.5, '3.5', 3.6, '3,6', 3.7, '3.7']
```
[the remark this answers is not in the document]
Comment (mraarif) [reply tests/test_travis_modernizer.py:16]: if we're handling a case where a Travis file might have a python version in quotes I suggest we use regex for that

File: django3_codemods/config_tools/travis_modernizer.py [hunk 2]
```diff
@@ -37,44 +44,56 @@ def remove_django_envs(self):
...
+                        {env: env_list[env]} for env in env_list if not self.DJANGO_PATTERN.search(env)
+                    ]
                 self.travis_dict['env'] = updated_list
             return self.travis_dict
 
     def update_python_version(self):
         """
         Add py38, remove py36 if present.
         """
+        versions_to_remove = [3.5, '3.5', 3.6, '3.6', 3.7, '3.7']
```
Comment (mraarif) [inline django3_codemods/config_tools/travis_modernizer.py:69]: same here as I mentioned for the tests

File: tests/test_travis_modernizer.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (UsamaSadiq) [reply tests/test_travis_modernizer.py]: We have different scenarios where we can get either a list, float, or dict based on different Travis configurations. Keeping this approach allows us to handle all cases with a single check but by switching it to regex, we'll need to add additional checks to handle all those cases. So in my opinion, we should go with this approach.

File: django3_codemods/config_tools/travis_modernizer.py (hunk 2 above)
Comment (UsamaSadiq) [reply django3_codemods/config_tools/travis_modernizer.py:69]: Same reason as above. Also, in near future, we'll probably need to switch from Travis to Github Actions so we shouldn't spend any more time on these scripts.

**Review by UsamaSadiq — APPROVED** [review]

**Review by mzulqarnain1 — APPROVED** [review]
